Tribler/Core/Modules/restapi/upgrader_endpoint.py

```diff
@@ -0,0 +1,44 @@
+from __future__ import absolute_import
+
+import logging
+
+from twisted.web import resource
+
+import Tribler.Core.Utilities.json_util as json
+
+
+class UpgraderEndpoint(resource.Resource):
+    """
+    With this endpoint you can control DB upgrade process of Tribler.
+    """
+
+    def __init__(self, session):
+        resource.Resource.__init__(self)
+        self._logger = logging.getLogger(self.__class__.__name__)
+        self.session = session
+
+    def render_DELETE(self, _):
+        """
+        .. http:put:: /upgrader
+
+        A PUT request to this endpoint will skip the DB upgrade process, if it is running.
+
+            **Example request**:
+
+            .. sourcecode:: none
+
+                curl -X DELETE http://localhost:8085/upgrader
+
+            **Example response**:
+
+            .. sourcecode:: javascript
+
+                {
+                    "skip": True
+                }
+        """
+
+        if self.session.upgrader:
```

Comment:
You should return an error if the upgrader is not there (probably error `404` since a resource could not be found).

Reply:
Good point, I'll fix that.